pkg/kubelet/lifecycle/predicate.go

```diff
@@ -277,12 +277,13 @@ func generalFilter(pod *v1.Pod, nodeInfo *schedulerframework.NodeInfo) []Predica
 
 	// Check taint/toleration except for static pods
 	if !types.IsStaticPod(pod) {
-		_, isUntolerated := corev1.FindMatchingUntoleratedTaint(nodeInfo.Node().Spec.Taints, pod.Spec.Tolerations, func(t *v1.Taint) bool {
+		taint, isUntolerated := corev1.FindMatchingUntoleratedTaint(nodeInfo.Node().Spec.Taints, pod.Spec.Tolerations, func(t *v1.Taint) bool {
 			// Kubelet is only interested in the NoExecute taint.
 			return t.Effect == v1.TaintEffectNoExecute
 		})
 		if isUntolerated {
-			reasons = append(reasons, &PredicateFailureError{tainttoleration.Name, tainttoleration.ErrReasonNotMatch})
+			errReason := fmt.Sprintf("node(s) had untolerated taint {%s: %s}", taint.Key, taint.Value)
```

Comment:
```suggestion
			errReason := fmt.Errorf("%w: taint %s: %s", tainttoleration.ErrReasonNotMatch, taint.Key, taint.Value)
```

Reply:
`tainttoleration.ErrReasonNotMatch` seems to be a constant, and I'm not sure if there are other components that rely on this for their judgments